kore/src/main/scala/org/kframework/rewriter/Rewriter.scala

```diff
@@ -15,31 +16,49 @@ trait Rewriter {
   //  def step(k: K): K
 
   /**
-   * (disregard this javadoc comment for now)
-   * Takes one rewriting step.
-   * - for regular execution, it returns the next K or False (i.e. an empty Or)
-   * - for symbolic execution, it can return any formula with symbolic constraints
-   * - for search, it returns an Or with multiple ground terms as children
+   * (disregard this javadoc comment for now) Takes one rewriting step.
+   *   - for regular execution, it returns the next K or False (i.e. an empty Or)
+   *   - for symbolic execution, it can return any formula with symbolic constraints
+   *   - for search, it returns an Or with multiple ground terms as children
    */
   def execute(k: kore.K, depth: Optional[Integer]): RewriterResult
 
-
   def `match`(k: kore.K, rule: Rule): kore.K
 
-
   /**
    * Execute a search of the Transition System.
-   * @param initialConfiguration The configuration to begin searching from.
-   * @param depth No. of transitions to consider before termination (Depth of Tree to traverse). Empty represents unbounded.
-   * @param bound No. of states to consider as final results. Empty represents unbounded.
-   * @param pattern The rule (pattern + side condition) that we're trying to find a substitution for.
-   * @return A list of substitutions, denoting all the configurations matching the given rule.
+   * @param initialConfiguration
+   *   The configuration to begin searching from.
+   * @param depth
+   *   No. of transitions to consider before termination (Depth of Tree to traverse). Empty
+   *   represents unbounded.
+   * @param bound
+   *   No. of states to consider as final results. Empty represents unbounded.
+   * @param pattern
+   *   The rule (pattern + side condition) that we're trying to find a substitution for.
+   * @return
+   *   A list of substitutions, denoting all the configurations matching the given rule.
```

Comment:
I don't want to be pedantic, but for me, the comments should be above the definition they're referring to, so maybe moving them above would look nicer!

Reply:
Unfortunately, the only configuration we have here is
- `docstrings.wrap = yes`, which ensures docstring comments fit in `maxColumn` by fully reformatting them as above
- `docstrings.wrap = no`, which allows docstrings to be arbitrarily long and left unformatted

IMO we should keep everything under the `maxColumn`, so I don't think it's worth addressing this

Reply:
Got it! That is unfortunate indeed! Could we manually modify it to be above the definition in this case and check if it happens again somewhere? I think this is the most evident and significant case, but I can also be wrong!

Reply:
Actually on second thought, I'm strongly opposed to moving the commments above.

These are ScalaDoc comments which generate nice static documentation web pages, and the ScalaDoc parser treats anything following `@param paramName` as the documentation for the parameter.

If we move the comments above the `@param` tag, that just incorrectly associates it with the previous `@param` tag, and all of our documentation will be off-by-one in the generated web page.

Reply:
Got it! It isn't much readable on the code, but it wouldn't make sense indeed to the generated Scala Doc